api/python/provisioner/ui/validation.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+#
+import re
+import ipaddress
+
+
+class Validation:
+
+    @staticmethod
+    def ipv4(ip):
+        try:
+            value = None
+            result = True
+            if ip:
+                value = ipaddress.IPv4Address(ip)
+                # TODO : Improve logic internally convert ip to
+                # canonical forms.
+                if ip != str(value):
+                    result = False
+        except ValueError:
+            result = False
+        return result
+
+    @staticmethod
+    def hostname(hostname):
+        result = True
+        hostname_regex = r"^([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z]{2,6}$"
```

Comment:
how are we confident with such a regex? I guess there might be a set of cases we we don't expect for now. e.g. this one in [django](https://github.com/django/django/blob/6726d750979a7c29e0dd866b4ea367eef7c8a420/django/core/validators.py#L47-L49). Can't we use some python build-in module for that?

Reply:
not able to find better way to validate hostname. I don't think we have any inbuild python package for this.

Can you please suggest some better way for validation of hostname?

Reply:
I believe, what we are validating here is just that the hostname is actually in FQDN format with no captital letters or special characters.
Does this handle multiple '.' though?

Reply:
yes , it handle multiple dots.

Reply:
>Can you please suggest some better way for validation of hostname?

I provided more  comprehensive one from django. We can keep the current one for now but please add some TODO comment. thanks

Reply:
comment is addressed, Added TODO comment